### DISCUSSION TITLE
Ability to set border colour for windows with urgent hint

### DISCUSSION THREAD

**Discussion by singpolyma** [qtile/qtile#2908] [Ideas]
This is very useful to tell which of the windows on a busy screen are asking for your attention.

**Comment by m-col** [comment]
This would be pretty cool and have obvious utility. Border colours are controlled by the layouts as each layout can have different border configurations. For urgency to be displayed via border colour it's likely that each layout would need to be slightly altered to accept an urgent border colour option. PR welcome!

An alternative approach might be to use the `client_urgent_hint_changed` hook and manually change the border colour (untested):
```py
from libqtile import hook, qtile

@hook.subscribe.client_urgent_hint_changed
def _(win):
    if win.urgent and win is not qtile.current_window:
        win.paint_borders("ff0000", win.borderwidth)
```

**Reply by singpolyma** [reply]
I tried this just now (took me way too long because most of my terminals do demand_attention instead of urgent, which qtile treats like urgent for some purposes but not for running the hook...) -- it does cause the border to flash to the selected colour, but then the borders immediately return to their normal colour (this is using column layout).  Not sure what event causes that, but I guess it may need to be implemented inside the layouts somehow to keep the state if the borders are being repainted a lot.

**Reply by m-col** [reply]
>  Not sure what event causes that, but I guess it may need to be implemented inside the layouts somehow to keep the state if the borders are being repainted a lot.

I think if the focus changes it will go back to normal and this is likely happening upon click on a window. I think the layouts would need to handle this.

**Reply by singpolyma** [reply]
Oh, if focus changes *at all* and not just the focus of that window, then a full repaint of all borders happens?  That makes sense.

**Reply by m-col** [reply]
I believe most layouts do that, yep.
